### PR TITLE
feat: funding stream table added

### PR DESCRIPTION
**Pull request by AlexZorkin**
- funding stream table added with id, name, description
- project table foreign key to funding stream added (funding_stream_id)

### CODE REVIEW

File: schema/deploy/tables/funding_stream.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+end
+$grant$;
+
+comment on table cif.funding_stream is 'Table containing information about a CIF funding_stream';
+comment on column cif.funding_stream.id is 'Unique ID for the funding_stream';
+comment on column cif.funding_stream.name is 'Shortname of the funding stream program';
+comment on column cif.funding_stream.description is 'Description of the funding_stream';
+
+insert into cif.funding_stream (name, description) values ('IA', 'Innovation Accelerator'), ('EP', 'Emissions Performance');
```
Comment (pbastia) [inline schema/deploy/tables/funding_stream.sql:39]: we need EP to be id = 1 and IA to be id = 2, that's how the business refers to them right now - and how it's being used to generate the unique project IDs

File: schema/deploy/tables/project.sql [hunk 2]
```diff
@@ -5,7 +5,8 @@ begin;
 create table cif.project(
   id integer primary key generated always as identity,
   cif_identifier integer unique not null,
-  description varchar(10000) not null
+  description varchar(10000) not null,
+  funding_stream_id int not null references cif.funding_stream(id)
```
Comment (pbastia) [inline schema/deploy/tables/project.sql:9]: just nitpicking about consistency 

```suggestion
  funding_stream_id integer not null references cif.funding_stream(id)
```

File: schema/deploy/tables/project.sql [hunk 3]
```diff
@@ -5,7 +5,8 @@ begin;
 create table cif.project(
```
Comment (pbastia) [inline schema/deploy/tables/project.sql:5]: you could add a
```sql
-- requires: tables/funding_stream
```
comment to have sqitch enforce order ([example in ciip](https://github.com/bcgov/cas-ciip-portal/blob/6aef3cd1833ae44ee3230c0bcc17eb8eea392fe9/schema/deploy/tables/application_revision_status_001.sql#L2))

**Review by pbastia — CHANGES_REQUESTED** [review]

File: schema/deploy/tables/funding_stream.sql (hunk 1 above)
Comment (AlexZorkin) [reply schema/deploy/tables/funding_stream.sql]: I swapped the order so EP will have id = 1 and IA id = 2

File: schema/deploy/tables/project.sql (hunk 2 above)
Comment (AlexZorkin) [reply schema/deploy/tables/project.sql]: fixed

File: schema/deploy/tables/project.sql (hunk 3 above)
Comment (AlexZorkin) [reply schema/deploy/tables/project.sql:6]: added

File: schema/deploy/tables/funding_stream.sql [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+end
+$grant$;
+
+comment on table cif.funding_stream is 'Table containing information about a CIF funding_stream';
+comment on column cif.funding_stream.id is 'Unique ID for the funding_stream';
+comment on column cif.funding_stream.name is 'Shortname of the funding stream program';
+comment on column cif.funding_stream.description is 'Description of the funding_stream';
+
+insert into cif.funding_stream (name, description) values ('EP', 'Emissions Performance'), ('IA', 'Innovation Accelerator');
```
Comment (pbastia) [inline schema/deploy/tables/funding_stream.sql:39]: Last little bit: this could use a comment to why the order matters

Comment (AlexZorkin) [reply schema/deploy/tables/funding_stream.sql:41]: updated

**Review by pbastia — DISMISSED** [review]

**Review by pbastia — APPROVED** [review]

**Review by pbastia — APPROVED** [review]

**Pull request merged**